Release notes for v0.12.1-alpha (#1288)
spiceai/spiceai: docs/release_notes/v0.12.1-alpha.md

# Spice v0.12.1-alpha (May 6, 2024)

The v0.12.1-alpha release introduces a new Snowflake data connector, support for UUID and TimestampTZ types in the PostgreSQL connector, and improved error messages across all data connectors. The Clickhouse data connector enables data streaming by default. The public SQL interface now restricts DML and DDL queries. Additionally, accelerated tables now fully support NULL values, and issues with schema conversion in these tables have been resolved.

## Highlights

- **Snowflake Data Connector**: Initial support for [Snowflake](https://www.snowflake.com/) as a data source.

- **Clickhouse Data Streaming**: Enables data streaming by default, eliminating in-memory result collection.

- **Read-only SQL Interface**: Disables DML (INSERT/UPDATE/DELETE) and DDL (CREATE/ALTER TABLE) queries for improved data source security.

- **Error Message Improvements**: Improved the error messages for commonly encountered issues with data connectors.

> Accelerated Tables: Supports NULL values across all data types and fixes schema conversion errors for consistent type handling.

Reviewer: ```suggestion
- **Accelerated Tables**: Supports NULL values across all data types and fixes schema conversion errors for consistent type handling.
```

# Spice v0.12.1-alpha (May 6, 2024)

> The v0.12.1-alpha release introduces a new Snowflake data connector, support for UUID and TimestampTZ types in the Postgres data connector, and improved error messages for common data connector issues. Data streaming from Clickhouse is now enabled by default, and the public SQL interface now disables DML and DDL queries. Accelerated tables now support NULL values for all supported data types, and schema conversion for accelerated tables has been fixed.

Reviewer 2: ```suggestion
The v0.12.1-alpha release introduces a new Snowflake data connector, support for UUID and TimestampTZ types in the PostgreSQL connector, and improved error messages across all data connectors. The Clickhouse data connector enables data streaming by default. The public SQL interface now restricts DML and DDL queries. Additionally, accelerated tables now fully support NULL values, and issues with schema conversion in these tables have been resolved.
```

## Highlights

- **Snowflake Data Connector**: Initial support for [Snowflake](https://www.snowflake.com/) as a data source.

> - **Clickhouse Data Streaming**: Data is now streamed from Clickhouse by default instead of waiting for all results to be collected in memory first.

Reviewer 2: ```suggestion
- **Clickhouse Data Streaming**: Enables data streaming by default, eliminating in-memory result collection.
```

> - **Read-only SQL Interface**: The public SQL interface now disables DML (Data Manipulation Language, i.e. `INSERT/UPDATE/DELETE`) and DDL (Data Definition Language `CREATE TABLE/ALTER TABLE`) queries.

Reviewer 2: ```suggestion
- **Read-only SQL Interface**: Disables DML (INSERT/UPDATE/DELETE) and DDL (CREATE/ALTER TABLE) queries for improved data source security.
```

- **Error Message Improvements**: Improved the error messages for commonly encountered issues with data connectors.

> - **Accererated Tables Support NULL Values**: Accelerated tables now support handling `NULL` values for all supported data types.
> 
> - **Fixes Accelerated Table Schema Conversion**: When the type of the accelerated engine for a given dataset is different than the same federated table, Spice now correctly converts the schema to match the accelerated engine. Previously the schema of the accelerated table engine was used, which led to hard-to-debug issues around type conversion that are now fixed.

Reviewer 2: ```suggestion
Accelerated Tables: Supports NULL values across all data types and fixes schema conversion errors for consistent type handling.
```
